samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -330,6 +397,28 @@ static void start(void)
 	return;
 }
 
+static void update_peer_is_rpa(enum peer_rpa new_peer_rpa)
+{
+	peer_is_rpa[cur_identity] = new_peer_rpa;
+
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		/* Assuming ID is written using only one digit. */
+		__ASSERT_NO_MSG(cur_identity < 10);
+		char key[MAX_KEY_LEN];
+		int err = snprintk(key, sizeof(key), MODULE_NAME "/%s%d",
+				   PEER_IS_RPA_STORAGE_NAME, cur_identity);
+
+		if ((err > 0) && (err < MAX_KEY_LEN)) {
+			err = settings_save_one(key, &peer_is_rpa[cur_identity],
```

Comment:
there is a possibility that this will not match a stored bond; this can happen when device will reset after bond is erased but before this value is updated + device will reset again after bonds are created but before the setting is stored.

There is little we can do about it and the only consequence will be usage of different adv technique (which should be fine anyway).